src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_FILED_DEFAULT = '',
+  SORT_FILED_NAME = 'name',
+  SORT_FILED_LENGTH = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: { sortField: SortType, isReversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_FILED_NAME:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_FILED_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
```

Comment:
It would be great to move switch...case out of the sort callback so there is no need to extra call it on default case